safety_prototype/README.rst

```diff
@@ -0,0 +1,64 @@
+README
+======
+
+Overview
+--------
+
+This is a prototype to implement `error handling`_ in Python.
+
+The user only calls API functions or methods, never SDK functions or methods.
+
+Every SDK must implement every public function or method defined in the API
+except for the SDK setting function.
+
+A safety mechanism is applied to every API function or method (except for the
+SDK setting function or method) and only to every API function or method.
+
+This safety mechanism has 3 main components:
+
+1. A predefined value to be returned in case of an exception being raised. This
+   value is predefined and independently set for every function or method.
+2. A `try` / `except` block that catches any exception raised when executing
+   the function or method code.
+3. A Python `warning`_ that is "raised" if an exception is raised in the code
+   protected by the safety mechanism.
+
+The API provides a function to set a specific SDK. This function is
+intentionally not protected by the safety mechanism mentioned above because the
+specification `mentions`_ this:
+
+    The API or SDK may fail fast and cause the application to fail on
+    initialization...
+
+*Initialization* is understood as the process of setting the SDK.
```

Comment:
Well, but how would this work? Imagine this situation:

```
# application.py

set_sdk("sdk")  # This can fail fast

...  # Here the user does a lot of stuff including using the OpenTelemetry API

meter = create_meter(...)  # This can also fail fast?
```

That would mean that after a lot of code has been executed something suddenly fails because creating a meter is considered initialization. This behaves like something that breaks the application code. Now, someone may argue that meters, tracers, etc are always created before doing anything else but there is no guarantee that would be the case. We could make it configurable that the creating of these objects fail fast but I see almost no value in this, the moment one of these objects is created and something goes wrong a warning will be raised and the user will know and if they want the application to crash they can run it in error mode. I don't see why we need to handle the creation of these objects in a special way.